wordpress-mobile/WordPress-FluxC-Android
# [Coupons] Update coupons

This PR adds the update coupon functionality & the example app test UI. 

Sorry about the large PR, there are many properties in coupons so that made the UI code lengthy. I will add unit tests later on separately.

![image](https://user-images.githubusercontent.com/1522856/165620374-1d7f4787-5e67-4d7c-a3f8-9f71d907042b.png)
![image](https://user-images.githubusercontent.com/1522856/165620686-38b58876-8720-4528-922a-72324b7fff8f.png)
![image](https://user-images.githubusercontent.com/1522856/165620765-f1c2da8d-7039-4912-99e0-61e6dbf49795.png)

**To test:**
1. Go to Woo -> Products
2. Click on Fetch products
3. Click on Fetch product categories
4. Navigate back to Woo -> Coupons
5. Select a site
6. Click on Update coupon
7. Click on Enter coupon ID, enter a valid coupon ID number and wait for a coupon to load
8. **Verify that all data is correctly loaded**
9. Modify different properties
10. **Tap on the check mark button and verify the data is correctly updated**
11. Tap on the Select expiry date
12. **Verify a date selection dialog appears**
13. Select a date
14. **Verify a correct date appears**
15. Tap on the Incl. and Excl. products
16. **Verify a dialog appears with a list of products**
17. Select different items and tap OK
18. **Verify the IDs are displayed correctly**
19. Do the same for incl. and excl. categories
20. **Update the coupon again and verify the products and categories are updated correctly**

**Note:**
After this PR is merged, the corresponding [Woo PR](https://github.com/woocommerce/woocommerce-android/pull/6419) must be also updated and merged to fix the build.
